lib/Mojo/WebSocketProxy/Backend.pm

```diff
@@ -1,20 +1,45 @@
 package Mojo::WebSocketProxy::Backend;
-# TODO(leonerd): Split out JSON-RPC-specific parts into a subclass
 
 use strict;
 use warnings;
 
-use Mojo::Base -base;
+no indirect;
 
-use MojoX::JSON::RPC::Client;
-use Guard;
+use Mojo::Util qw(class_to_path);
 
 ## VERSION
 
-has 'url';
+our %CLASSES = (
+    jsonrpc   => 'Mojo::WebSocketProxy::Backend::JSONRPC',
+    job_async => 'Mojo::WebSocketProxy::Backend::JobAsync',
+);
+
+sub class_for_type {
+    my ($class, $type) = @_;
+    my $target_class = $CLASSES{$type};
+    # extra () around require to avoid treating the function call as a class name
+    require(class_to_path($target_class)) unless $target_class->can('new');
```

Comment:
Either solution suggested above then avoids this problem, as `Module::PluginFinder` would find the packages, or the subclass-registers-itself just needs the containing program to `use` all the appropriate modules which then register themselves as a load-time side-effect.

Reply:
Done 99a543c